### ISSUE TITLE
Wrong error message in a query with combination function over empty lookup

### ISSUE THREAD

**Issue by starstuck** [dalmatinerdb/dqe#43]
Imagine that indexer have already indexed `some_known_metric`, but does not know nothing about `some_unknown_metric`.

Now if you run a combined query:

```
SELECT avg(some_known_metric FROM bucket, some_unknown_metric FROM bucket)
```

it will work as one would expect. You will get average across all known metric.

But if you try to combine metrics that are not looked up properly

```
SELECT avg(some_unknown_metric FROM bucket)
```

You will get very confusing error message:

```
Combination functions can't have mix resolutions as children.
```

Which is not what one would expected, especially that aggregation functions in the same situation will throw message about missing data.

I think we should provide more consistent error message.

I manage to track down the error down to 'get_times_' function in https://github.com/dataloop/dqe/blob/master/src/dql_resolution.erl#L96. Because lookup returns empty 'Elements' list, 'Rmss' will be empty and thus not match case condition.

I am not entirely sure what we should do in this particular case. If we wanted to have it consistent with other parts of query system, we should change 'dql_resolution' module to not throw error, but carry on, just omitting empty execution branch.

On the other hand it would also make sense to treat failed lookup as an error and raise it early in 'dql_expand' module. 'Dql' already seems to be treating situation when no data can be fetch as an error. Raising those errors early, during expansion phase would allow us to provide more context, with details which lookups are failing, thus assisting user in fixing query.

I have spent already fair amount of time investigating the issue, so I can prepare some fix. I just need some advice which direction we should go and what would make more sense for architectural point of view.

**Comment by Licenser** [comment]
Heya,
that's a good catch! Both solutions are valid in one way or another, but I'd feel better with failing. Hard stop seems nicer then quiet omission especially when it's a typo and not just something that isn't there.

Taking the first example

`SELECT avg(some_known_metric FROM bucket, some_other_metirc FROM bucket)` would really be better to fail then quietly ignore data from `some_other_metric`.

**Comment by starstuck** [comment]
Agree with you! I will prepare some pull request in line with that strategy.

Cheers

**Comment by Licenser** [comment]
awesome :D

**Comment by starstuck** [comment]
Resolved by #45
